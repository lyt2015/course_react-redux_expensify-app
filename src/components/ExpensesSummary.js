import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'

import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <h1>
      Viewing {expenseCount} expense{expenseCount === 1 ? '' : 's'} totalling{' '}
      {numeral(expensesTotal / 100.0).format('$0,0.00')}
    </h1>
  </div>
)

const mapStateToProps = state => {
  const expenses = selectExpenses(state.expenses, state.filters)

  return {
    expenseCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses),
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
