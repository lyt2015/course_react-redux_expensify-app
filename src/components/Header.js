import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'

import { startLogout } from '../actions/auth'

// eslint-disable-next-line no-shadow
export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active">
      Dashboard
    </NavLink>
    <br />
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
    <br />
    <NavLink to="/help" activeClassName="is-active">
    Help
    </NavLink>
    <br />
    <button onClick={startLogout}>Logout</button>
  </header>
)

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
})

export default connect(
  undefined,
  mapDispatchToProps
)(Header)
