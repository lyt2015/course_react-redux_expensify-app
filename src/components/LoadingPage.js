import React from 'react'

const LoadingPage = () => (
  <div className="loader">
    <img className="loader__image" alt="now loading" src="/images/loader.gif" />
  </div>
)

export default LoadingPage
