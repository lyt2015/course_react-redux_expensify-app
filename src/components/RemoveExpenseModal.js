import React from 'react'
import Modal from 'react-modal'

Modal.setAppElement(document.getElementById('app'))

const RemoveExpenseModal = props => (
  <Modal
    isOpen={props.isOpen}
    onRequestClose={props.handleCloseModal}
    contentLabel="Delete Expense?"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Delete Expense?</h3>
    {<p className="modal__body">{props.selectedExpense}</p>}
    <div className="modal__button">
      <button className="button" onClick={props.confirmRemove}>
        Yes
      </button>
      <button className="button" onClick={props.handleCloseModal}>
        No
      </button>
    </div>
  </Modal>
)

export default RemoveExpenseModal
