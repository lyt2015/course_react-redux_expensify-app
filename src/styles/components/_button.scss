.button {
  background: $dark-blue;
  border: none;
  color: $dark-white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.2s, color 0.2s;
}

.button:hover {
  background: $blue;
  color: white;
}

.button--link,
.button--link:hover {
  background: none;
}

.button--secondary {
  @extend .button;
  background: #888;
}
