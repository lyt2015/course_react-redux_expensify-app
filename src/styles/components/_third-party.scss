.SingleDatePickerInput {
  width: 100%;
  text-align: center;
}
