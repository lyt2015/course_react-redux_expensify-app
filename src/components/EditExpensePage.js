import React from 'react'
import { connect } from 'react-redux'

import ExpenseForm from './ExpenseForm'
import RemoveExpenseModal from './RemoveExpenseModal'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  state = {
    isModalOpen: false,
  }

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }

  confirmRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.handleCloseModal()
    this.props.history.push('/')
  }

  onRemove = () => {
    this.setState(() => ({ isModalOpen: true }))
  }

  handleCloseModal = () => {
    this.setState(() => ({ isModalOpen: false }))
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="button--secondary" onClick={this.onRemove}>
            Remove Expense
          </button>
        </div>
        <RemoveExpenseModal
          isOpen={this.state.isModalOpen}
          handleCloseModal={this.handleCloseModal}
          selectedExpense={this.props.expense.description}
          confirmRemove={this.confirmRemove}
        />
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(({ id }) => id === props.match.params.id),
})

const mapDispatchToProps = dispatch => ({
  startEditExpense: (id, expense) => {
    dispatch(startEditExpense(id, expense))
  },
  startRemoveExpense: data => {
    dispatch(startRemoveExpense(data))
  },
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage)
