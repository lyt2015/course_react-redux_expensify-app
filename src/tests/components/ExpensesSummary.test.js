import React from 'react'
import { shallow } from 'enzyme'

import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render ExpensesSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={10000} />)

  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={7} expensesTotal={77777} />)

  expect(wrapper).toMatchSnapshot()
})
