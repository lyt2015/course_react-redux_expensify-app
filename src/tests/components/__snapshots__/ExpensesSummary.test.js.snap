// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    7
     expense
    s
     totalling
     
    $777.77
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling
     
    $100.00
  </h1>
</div>
`;
